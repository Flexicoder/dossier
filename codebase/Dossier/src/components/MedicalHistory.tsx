import React, { Component } from 'react';
import { TextInput, View } from 'react-native';
import  styles  from '../assets/styles'

interface MedicalHistoryProps {
    details?: string | null;
}

export default class MedicalHistory extends Component<MedicalHistoryProps> {
    constructor(props: MedicalHistoryProps) {
        super(props)
        this.state = { details: props.details }
    }

    render() {
        return (
        <View style={styles.container}>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                        <TextInput
                            placeholder="Details"
                            multiline={true}
                            onChangeText={text => this.setState({ details: text})}
                            defaultValue={this.props.details || ""} />
                    </View>
                </View>
            </View>
        );
    }
}