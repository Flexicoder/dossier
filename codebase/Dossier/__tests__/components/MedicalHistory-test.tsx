
import 'react-native';
import React from 'react';
import MedicalHistory from '../../src/components/MedicalHistory';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe("Medical History", () => {
    it('renders correctly', () => {
        const medicalHistory = renderer.create(<MedicalHistory />);
        expect(medicalHistory.toJSON()).toMatchSnapshot();
      });
})


describe("Medical History", () => {
    it('contains supplied data', () => {
        const medicalHistory = renderer.create(<MedicalHistory details="Asthma" />);
        const stringVersion = JSON.stringify(medicalHistory.toJSON())
        expect(stringVersion).toMatch(/Asthma/);
      });
})