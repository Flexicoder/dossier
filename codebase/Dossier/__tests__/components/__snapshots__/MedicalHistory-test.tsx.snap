// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Medical History renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        multiline={true}
        onChangeText={[Function]}
        placeholder="Details"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
